clouddriver-web/src/test/groovy/com/netflix/spinnaker/clouddriver/requestqueue/pooled/PooledRequestQueueSpec.groovy

```diff
@@ -35,12 +37,51 @@ class PooledRequestQueueSpec extends Specification {
 
   def "should time out if request does not complete"() {
     given:
-    def queue = new PooledRequestQueue(new NoopRegistry(), 10, 1)
+    def queue = new PooledRequestQueue(new NoopRegistry(), 5000, 10, 1)
 
     when:
-    Long result = queue.execute("foo", { Thread.sleep(20); return 12345L })
+    queue.execute("foo", { Thread.sleep(20); return 12345L })
 
     then:
     thrown(PromiseTimeoutException)
   }
+
+
+  def "should time out if request does not start in time"() {
```

Comment:
How long will that take to run? Seems like the executor could evict that 40 second sleep thread in cleanup so it's not hanging around.